torch/_inductor/compile_fx.py

```diff
@@ -261,11 +261,21 @@ def _resolve_name_collision(mod: GraphModule, gm: GraphModule) -> None:
     with a new number post fix.
     """
 
+    existing_keys = OrderedSet(
+        [name for name, val in mod.named_parameters(remove_duplicate=False)]
+    )
+    existing_keys.update(
+        OrderedSet([name for name, val in mod.named_buffers(remove_duplicate=False)])
+    )
+
     def find_smallest_i(graph: fx.Graph, prefix: str) -> int:
         i = 0
         for node in graph.nodes:
             if node.op == "get_attr" and node.target.startswith(prefix):
                 i = max(i, int(node.target.split(prefix)[-1]))
+        for key in existing_keys:
+            if key.startswith(prefix):
+                i = max(i, int(key.split(prefix)[-1]))
```

Comment:
I just realized, what if the target is like "tensor_constant_foo"? wouldn't the `int(node.target.split(prefix)[-1])` fail?

Reply:
@angelayi  I don't think this will happen, because we only call `find_smallest_i` when there is a name collision, and name collision only happens when `aot_export_module` create attribute names.

`aot_export_module` only create attribute names that are like `tensor_constant_<int>`. 

I can change this to be more robust though.

Reply:
@angelayi Fixed to be more robust now. If the post_fix is not a digit, we'll just skip it when finding the next number for `i`.